client/src/context.rs

```diff
@@ -161,21 +172,20 @@ impl Context for ClientContext {
             Ok(mut v) => v.drain(..).map(|i| i.into()).collect(),
             Err(e) => return Err(e)
         };
-        let vs = v.into_boxed_slice();
-        let coll = ffi::cubeb_device_collection {
-            count: vs.len(),
-            device: vs.as_ptr()
-        };
+        let mut vs = v.into_boxed_slice();
+        let coll = unsafe { &mut *collection.as_ptr() };
+        coll.device = vs.as_mut_ptr();
+        coll.count = vs.len();
         // Giving away the memory owned by vs.  Don't free it!
         // Reclaimed in `device_collection_destroy`.
         mem::forget(vs);
-        Ok(coll)
+        Ok(())
     }
 
-    fn device_collection_destroy(&self, collection: *mut ffi::cubeb_device_collection) {
+    fn device_collection_destroy(&mut self, collection: &DeviceCollectionRef) -> Result<()> {
```

Comment:
I guess this is a question for the cubeb-rs change, but I missed it there: doesn't this leave DeviceCollectionRef in an unsafe state?

Reply:
Yup. Fixed it.